### PR TITLE
[ONNX] Add ReduceLogSum-11,13,18

### LINKED ISSUE #20561: [Good First Issue]: Align behavior of ONNX Frontend function ReduceLogSum-11, 13, 18 with original framework
### Context

Neural networks are graphs consisting of nodes called operators. Each operator corresponds to a mathematical function, usually described in framework's documentation or an AI standard, such as ONNX.
OpenVINO ONNX Frontend is a component responsible for working with ONNX graphs and requires implementation of different ONNX operators in order to use ONNX models.
This task requires alignment between OpenVINO ONNX Frontend and original framework implementations of ReduceLogSum for next list of opsets: [opset 11](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceLogSum-11),  [opset 13](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceLogSum-13),  [opset 18](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceLogSum-18)
Necessary help will be provided by ONNX Fronted team.

### What needs to be done?

First of all, please, take a look on [ReduceMax](https://github.com/openvinotoolkit/openvino/pull/23475) PR for a reference.

Operator details can be found in [ONNX Operators](https://github.com/onnx/onnx/blob/main/docs/Operators.md#ReduceLogSum)
More details can be found in ONNX Changelog: [opset 11](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceLogSum-11),  [opset 13](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceLogSum-13),  [opset 18](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceLogSum-18)
1. Function already has a common implementation in OpenVINO. First of all, you need to review a documentation and prepare a table with differences between versions. It could be, for instance, a missing property, extended/reduced coverage of existing property, etc...
2. Copy existing implementation [here](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/frontend/src/op) to make it aligned with original framework (extend or reduce coverage of a common implementation). Copy of modified implementation should be in a defined opset, or in opset 1 in case it implements oldest implementation. [Example of multi-opset operation](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/softmax.cpp#L50).
3. Register the function in [ops_bridge.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/ops_bridge.cpp#L554-L560) while keeping alphabetical order
4. Create test model(s) in [ONNX models directory](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/tests/models/). OpenVINO test infrastructure then converts prototxt files to ONNX models - you will use those models later in tests
5. Add tests covering all use cases [here](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/onnx_import.in.cpp)
6. Check [Python xfailed tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/tests_python/test_backend.py) to find a test marked as a xfailed for added functionality. If any exist - remove corresponding lines and try to verify by using cmdline "python -m pytest -k name_of_test".
More details in [adding operators to ONNX Frontend guide](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/how_to_add_op.md)

### Example Pull Requests

_No response_

### Resources

- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Intel DevHub Discord channel](https://discord.gg/wPuqAujS) - engage in discussions, ask questions and talk to OpenVINO developers
- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [User documentation](https://docs.openvino.ai/)
- Guides on [ONNX Frontend architecture](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/architecture.md) and [ONNX Frontend tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/tests.md) are also available
- You can also see a [blog post on contributing to OpenVINO](https://medium.com/openvino-toolkit/how-to-contribute-to-an-ai-open-source-project-c741f48e009e)

### Contact points

@gkrivor 

### Ticket

_No response_

### PR DESCRIPTION
**Pull request by mory91**
### Details:
 - Extended ReduceLogSum by opsets 11,13,18

### Tickets:
 - Closes #20561

### CODE REVIEW

File: src/frontends/onnx/tests/onnx_import.in.cpp [hunk 1]
```diff
@@ -910,6 +910,22 @@ OPENVINO_TEST(${BACKEND_NAME}, onnx_model_reduce_log_sum) {
     test_case.run();
 }
 
+OPENVINO_TEST(${BACKEND_NAME}, onnx_model_reduce_log_sum_18) {
+    auto model = convert_model("reduce_log_sum_18.onnx");
+
+    // input data shape (1, 1, 4, 4)
+    Inputs inputs{
+        ov::test::NDArray<float, 4>({{{{1, 1, 1, 1}, {1, 1, 1, 1}, {1, 1, 1, 1}, {1, 1, 1, 1}}}}).get_vector()};
```
Comment (mitruska) [inline src/frontends/onnx/tests/onnx_import.in.cpp:918]: For the default `axes` such input filled with ones can be ok, but for tests with `axes` provided, it would be better to fill the first input with different values.

**Review by mitruska — COMMENTED** [review]
Hello @mory91, thank you for this contribution!
The import code looks logically correct, I've suggested some improvements to avoid code redundancy.

File: src/frontends/onnx/frontend/src/op/reduce.cpp [hunk 2]
```diff
@@ -174,13 +181,22 @@ ov::OutputVector reduce_sum_square(const ov::frontend::onnx::Node& node) {
 */
 
 namespace set_13 {
+ov::OutputVector reduce_log_sum(const ov::frontend::onnx::Node& node) {
+    const ov::Output<ov::Node> sum_node =
+        make_ov_reduction_op<v1::ReduceSum>(node, node.get_ov_inputs().at(0), supported_types_v2);
```
[the remark this answers is not in the document]
Comment (mory91) [reply src/frontends/onnx/frontend/src/op/reduce.cpp]: I applied this idea to the set_1, if the graph the same, I think it would be ok.

File: src/frontends/onnx/tests/onnx_import.in.cpp (hunk 1 above)
Comment (mory91) [reply src/frontends/onnx/tests/onnx_import.in.cpp:918]: I also added one more test for the axes

**Comment by mlukasze** [comment]
hey @mory91 please correct formatting to pass clang tests

**Comment by mory91** [comment]
@mlukasze Fixed

**Comment by mory91** [comment]
@mlukasze is there any way that triggering the CI wouldn't need additional confirmation for this pull request? Anyways, I fixed the latest styling error which was produced with my latest added test.

**Comment by mlukasze** [comment]
> @mlukasze is there any way that triggering the CI wouldn't need additional confirmation for this pull request? Anyways, I fixed the latest styling error which was produced with my latest added test.

I ran 1 set of tests for you, reviewer will decide to enable full scope later.

**Comment by mory91** [comment]
The tests were passed (Unless windows tests).

File: src/frontends/onnx/tests/models/reduce_log_sum_18_axes_as_input.prototxt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+          }
+          dim {
+            dim_value: 1
+          }
+          dim {
+            dim_value: 1
+          }
+          dim {
+            dim_value: 4
+          }
```
Comment (mitruska) [inline src/frontends/onnx/tests/models/reduce_log_sum_18_axes_as_input.prototxt:63]: Output shape inference is independent from the final output shape defined in the onnx model, but to be consistent with the expected `Shape{1, 1, 4, 1}` test, it would be better to align or define the model output in .prototxt as dynamic.
```suggestion
      tensor_type {
        elem_type: 1
        shape {
          dim {
            dim_value: 1
          }
          dim {
            dim_value: 1
          }
          dim {
            dim_value: 4
          }
          dim {
            dim_value: 1
          }
```

**Review by mitruska — COMMENTED** [review]
Hello @mory91, thank you for applying the comments, the changes LGTM.
Please sync branch with the latest master, and resolve merge conflicts.
Then we will run full validation cycle.

**Comment by mory91** [comment]
@mitruska thanks for the review, 
resolved.

File: src/frontends/onnx/frontend/src/op/reduce.cpp [hunk 4]
```diff
@@ -205,6 +204,10 @@ ov::OutputVector reduce_max(const ov::frontend::onnx::Node& node) {
 }  // namespace set_13
 
 namespace set_18 {
+ov::OutputVector reduce_log_sum(const ov::frontend::onnx::Node& node) {
+    const ov::Output<ov::Node> sum_node =
+        make_ov_reduction_op<v1::ReduceSum>(node, node.get_ov_inputs().at(0), supported_types_v2, false);
+    return {std::make_shared<v0::Log>(sum_node)};
```
Comment (mitruska) [inline src/frontends/onnx/frontend/src/op/reduce.cpp:210]: Merge conflicts resolved, but closing bracket is missing:
```suggestion
ov::OutputVector reduce_log_sum(const ov::frontend::onnx::Node& node) {
    const ov::Output<ov::Node> sum_node =
        make_ov_reduction_op<v1::ReduceSum>(node, node.get_ov_inputs().at(0), supported_types_v2, false);
    return {std::make_shared<v0::Log>(sum_node)};
}
```

**Comment by mitruska** [comment]
build_jenkins

File: src/frontends/onnx/tests/onnx_import.in.cpp [hunk 5]
```diff
@@ -910,6 +910,35 @@ OPENVINO_TEST(${BACKEND_NAME}, onnx_model_reduce_log_sum) {
...
+
+OPENVINO_TEST(${BACKEND_NAME}, onnx_model_reduce_log_sum_18_axes_as_input) {
+    auto model = convert_model("reduce_log_sum_18_axes_as_input.onnx");
+    auto test_case = ov::test::TestCase(model, s_device);
+
+    test_case.add_input<float>(Shape{1, 1, 4, 4}, {2, 1, 4, 2, 3, 1, 3, 2, 4, 2, 4, 2, 2, 2, 1, 4});
+    test_case.add_input<int64_t>({3});
+
+    test_case.add_expected_output(Shape{1, 1, 4, 1},
+                                  std::vector<float>{2.19722458, 2.19722458, 2.48490665, 2.19722458});
```
Comment (mitruska) [inline src/frontends/onnx/tests/onnx_import.in.cpp:937]: Message from the [windows CI log](https://github.com/openvinotoolkit/openvino/actions/runs/8379345729/job/22946136029?pr=23508#step:15:6469):

`openvino_build\src\frontends\onnx\tests\onnx_import_INTERPRETER.cpp(937): warning C4305: 'initializing': truncation from 'double' to '_Ty'
`

The following change should fix the issue:

```suggestion
                                  std::vector<float>{2.19722458f, 2.19722458f, 2.48490665f, 2.19722458f});
```

**Review by mitruska — COMMENTED** [review]
One more hint to fix the failing CI checks.

File: src/frontends/onnx/tests/onnx_import.in.cpp (hunk 5 above)
Comment (mory91) [reply src/frontends/onnx/tests/onnx_import.in.cpp:937]: Thank you

**Comment by mitruska** [comment]
build_jenkins

File: src/frontends/onnx/frontend/src/op/reduce.cpp [hunk 6]
```diff
@@ -141,7 +141,7 @@ std::shared_ptr<ov::Node> make_ov_reduction_op(const Node& node,
 namespace set_1 {
 ov::OutputVector reduce_log_sum(const ov::frontend::onnx::Node& node) {
     const ov::Output<ov::Node> sum_node =
-        make_ov_reduction_op<v1::ReduceSum>(node, node.get_ov_inputs().at(0), supported_types_v1);
+        make_ov_reduction_op<v1::ReduceSum>(node, node.get_ov_inputs().at(0), supported_types_v2);
```
Comment (gkrivor) [inline src/frontends/onnx/frontend/src/op/reduce.cpp:144]: Please, don't change it (https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceLogSum-1).

New types are available since opset-13 (https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceLogSum-13). Please, introduce it in opset-13.

File: src/frontends/onnx/frontend/src/ops_bridge.cpp [hunk 7]
```diff
@@ -480,6 +480,8 @@ OperatorsBridge::OperatorsBridge() {
     REGISTER_OPERATOR("RandomUniformLike", 1, random_uniform_like);
     REGISTER_OPERATOR("Reciprocal", 1, reciprocal);
     REGISTER_OPERATOR("ReduceLogSum", 1, reduce_log_sum);
+    register_operator("ReduceLogSum", VersionRange{1, 17}, op::set_1::reduce_log_sum);
```
Comment (gkrivor) [inline src/frontends/onnx/frontend/src/ops_bridge.cpp:483]: Great, thanks to use modern style! :) Do not forget to add opset-13 :)

**Review by gkrivor — CHANGES_REQUESTED** [review]
Hi @mory91 , thanks for your contribution!

I mostly agree with change, except one small detail. Please, fix it asap and I'm planning merge your PR!

File: src/frontends/onnx/frontend/src/op/reduce.cpp (hunk 6 above)
Comment (mitruska) [reply src/frontends/onnx/frontend/src/op/reduce.cpp:144]: We have already discussed it in the comments above, please take a look at https://github.com/openvinotoolkit/openvino/pull/23508#discussion_r1531005884.
The subgraph is the same, so in such case different types restrictions between onnx versions are artificial from the ov perspective, and lead to code redundancy.

File: src/frontends/onnx/frontend/src/ops_bridge.cpp (hunk 7 above)
Comment (mory91) [reply src/frontends/onnx/frontend/src/ops_bridge.cpp:483]: Doesn’t it already include 13? After a talk with @mitruska we decided to keep it this way since there are no differences in the subgraphs generated.

File: src/frontends/onnx/tests/tests_python/test_backend.py [hunk 8]
```diff
@@ -482,17 +482,12 @@ def expect_fail(test_case_path, xfail):  # type: (str) -> None
         "OnnxBackendNodeModelTest.test_reduce_l2_keep_dims_random_cpu",
         "OnnxBackendNodeModelTest.test_reduce_l2_negative_axes_keep_dims_example_cpu",
         "OnnxBackendNodeModelTest.test_reduce_l2_negative_axes_keep_dims_random_cpu",
-        "OnnxBackendNodeModelTest.test_reduce_log_sum_asc_axes_cpu",
```
Comment (gkrivor) [inline src/frontends/onnx/tests/tests_python/test_backend.py:485]: Regarding issues with tests. I took a look on a models. They fails for your particular case because axes has more than 1 item.

You can take a look on a source scripts: https://github.com/onnx/onnx/blob/0bb2775098cb72c5619451f37cc0c8f9f335893f/onnx/backend/test/case/node/reduce_log_sum.py#L14 it has two axes for reduction.

Currently, such tests are failing (known issue). It isn't issue in your code, but it is blocking merge.

We should return back these four tests:
```
        "OnnxBackendNodeModelTest.test_reduce_log_sum_asc_axes_cpu",
        "OnnxBackendNodeModelTest.test_reduce_log_sum_asc_axes_expanded_cpu",
        "OnnxBackendNodeModelTest.test_reduce_log_sum_desc_axes_cpu",
        "OnnxBackendNodeModelTest.test_reduce_log_sum_desc_axes_expanded_cpu",
```

**Review by gkrivor — APPROVED** [review]
I don't want to block you due to our internal discussion. I'll change code as I want further commit.

**Comment by mory91** [comment]
@gkrivor I see still two windows CI tests are failing, I read the logs and I couldn't see why its happening. I also don't have a windows machine to test it out myself :(

File: src/frontends/onnx/frontend/src/op/reduce.cpp [hunk 9]
```diff
@@ -208,6 +207,14 @@ namespace set_18 {
...
     return {make_ov_reduction_op<v1::ReduceMax>(node, node.get_ov_inputs().at(0), supported_types_v3, false)};
 }
+ov::OutputVector reduce_log_sum(const ov::frontend::onnx::Node& node) {
+    const ov::Output<ov::Node> sum_node =
+        make_ov_reduction_op<v1::ReduceSum>(node, node.get_ov_inputs().at(0), supported_types_v2, false);
+    return {std::make_shared<v0::Log>(sum_node)};
+}
+ov::OutputVector reduce_max(const ov::frontend::onnx::Node& node) {
+    return {make_ov_reduction_op<v1::ReduceMax>(node, node.get_ov_inputs().at(0), supported_types_v3, false)};
+}
```
Comment (mitruska) [inline src/frontends/onnx/frontend/src/op/reduce.cpp:217]: Looks like merge artifact, `reduce_max` is already defined above, so it currently appears twice within the `set_18` namespace:
https://github.com/openvinotoolkit/openvino/blob/484db1db51c9c4c1d6d7bafe7c57644a0283cc2e/src/frontends/onnx/frontend/src/op/reduce.cpp#L207-L209
Please remove the reduce_max duplication:
```suggestion
```

**Review by mitruska — COMMENTED** [review]
Merge artifact to be resolved

File: src/frontends/onnx/frontend/src/op/reduce.cpp (hunk 9 above)
Comment (mory91) [reply src/frontends/onnx/frontend/src/op/reduce.cpp]: removed thank you

**Comment by mitruska** [comment]
Current CI issues are not related to the PR changes, it should be fixed soon, no action required so far.

**Comment by mitruska** [comment]
build_jenkins
